ImperatorToCK3/Configuration.cs

```diff
@@ -0,0 +1,126 @@
+    class Configuration : Parser
+    {
+        public string SaveGamePath { get; internal set; } = "";
+        public string ImperatorPath { get; internal set; } = "";
+        public string ImperatorModsPath { get; internal set; } = "";
+        public string Ck3Path { get; internal set; } = "";
+        public string Ck3ModsPath { get; internal set; } = "";
+        public string OutputModName { get; internal set; } = "";
+        public IMPERATOR_DE_JURE ImperatorDeJure { get; internal set; } = IMPERATOR_DE_JURE.NO;
+        public bool ConvertBirthAndDeathDates { get; internal set; } = true;
+
+        public Configuration()
+        {
+            Logger.Log(LogLevel.Info, "Reading configuration file");
+            RegisterKeys();
+            ParseFile("configuration.txt");
+            ClearRegisteredRules();
+            SetOutputName();
+            VerifyImperatorPath();
+            VerifyCK3Path();
+        }
+
+        void RegisterKeys()
+        {
+            RegisterKeyword("SaveGame", (sr) =>
+            {
+                SaveGamePath = new SingleString(sr).String;
+                Logger.Log(LogLevel.Info, "Save game set to: " + SaveGamePath);
+            });
+            RegisterKeyword("ImperatorDirectory", (sr) =>
+            {
+                ImperatorPath = new SingleString(sr).String;
+            });
+            RegisterKeyword("ImperatorModsDirectory", (sr) =>
+            {
+                ImperatorModsPath = new SingleString(sr).String;
+            });
+            RegisterKeyword("CK3directory", (sr) =>
+            {
+                Ck3Path = new SingleString(sr).String;
+            });
+            RegisterKeyword("CK3ModsDirectory", (sr) =>
+            {
+                Ck3ModsPath = new SingleString(sr).String;
+            });
+            RegisterKeyword("output_name", (sr) =>
+            {
+                OutputModName = new SingleString(sr).String;
+                Logger.Log(LogLevel.Info, "Output name set to: " + OutputModName);
+            });
+            RegisterKeyword("ImperatorDeJure", (sr) =>
+            {
+                var deJureString = new SingleString(sr).String;
+                try
+                {
+                    ImperatorDeJure = (IMPERATOR_DE_JURE)Convert.ToInt32(deJureString);
+                    Logger.Log(LogLevel.Info, "ImperatorDeJure set to: " + deJureString);
+                } catch(Exception e)
+                {
+                    Logger.Log(LogLevel.Error, "Undefined error, ImperatorDeJure value was: " + deJureString + "; Error message: " + e.ToString());
+                }
+            });
+            RegisterKeyword("ConvertCharacterBirthAndDeathDates", (sr) =>
+            {
+                var valStr = new SingleString(sr).String;
+                if (valStr == "true")
+                {
+                    ConvertBirthAndDeathDates = true;
+                }
+                else if (valStr == "false")
```

Comment:
Change to a straight else, or add an else condition to handle this being set to something strange.